pkg/common/common_test.go

```diff
@@ -411,3 +413,249 @@ func TestMergeMapStringString(t *testing.T) {
 		})
 	}
 }
+
+func TestUnMarshallLimitNamespace(t *testing.T) {
+	testCases := []struct {
+		name           string
+		namespace      string
+		expectedKey    client.ObjectKey
+		expectedDomain string
+		expectedError  bool
+	}{
+		{
+			name:           "when namespace is valid and contains both namespace and domain then return the correct values",
+			namespace:      "exampleNS/exampleGW#domain.com",
+			expectedKey:    client.ObjectKey{Name: "exampleGW", Namespace: "exampleNS"},
+			expectedDomain: "domain.com",
+			expectedError:  false,
+		},
+		{
+			name:           "when namespace is invalid (no '#domain') then return an error",
+			namespace:      "exampleNS/exampleGW",
+			expectedKey:    client.ObjectKey{},
+			expectedDomain: "",
+			expectedError:  true,
+		},
+		{
+			name:           "when namespace contains more than one '#' then return an error",
+			namespace:      "exampleNS/exampleGW#domain.com#extra",
```

Comment:
@eguzki, @didierofrivia Are we considering this case? 
For this, I have a quick fix, adding `strings.ContainsRune(ns[delimIndex+1:], '#')` to the first if-clause:

```
func UnMarshallLimitNamespaceMultipleHashtags(ns string) (client.ObjectKey, string, error) {
	delimIndex := strings.IndexRune(ns, '#')
	if delimIndex == -1 || strings.ContainsRune(ns[delimIndex+1:], '#') {
		return client.ObjectKey{}, "", fmt.Errorf("failed to split on #")
	}
        // rest of the code
}
```

Reply:
After negotiating, this test-case was decided to be deleted, because it is on Kubernetes to validate it.
Thanks, @eguzki and @didierofrivia :angel: